app/redux/selectors/AnalysisSelectors.js

```diff
@@ -111,6 +113,10 @@ export const getDidUpdateMetric = (state, set) => !areEqual(set.metric, getOrigM
 
 export const getDidUpdateTags = (state, set) => !areArraysEqual(set.tags, getOrigTags(state));
 
+export const getDidDeleteSet = (state, set) => (Boolean(set.deleted) !== (getOrigDeletedFlag(state))) && !getOrigDeletedFlag(state);
+
+export const getDidRestoreSet = (state, set) => (Boolean(set.deleted) !== (getOrigDeletedFlag(state))) && getOrigDeletedFlag(state);
+
```

Comment:
- [ ] does this work if the set doesn't HAVE a deleted flag?

Reply:
Yes, as we are checking the Boolean of set.deleted, so undefined is false.

Reply:
- [ ] However, a set is considered deleted if it's empty right? So you'd be deleting a deleted set no?

Reply:
Updating to account for .removed